url89cafed56a_AlexDiru_esper_compiler_final_tgz-pJ8-compiler_ProgramJ8/src/compiler/EsperLexer.java

```diff
@@ -0,0 +1,1613 @@
+// $ANTLR 3.4 C:\\Users\\Alex\\Dropbox\\EclipseWorkspace\\esper-compiler\\src\\compiler\\Esper.g 2013-03-15 18:17:43
+
+  package compiler;
+
+
+import org.antlr.runtime.*;
+import java.util.Stack;
+import java.util.List;
+import java.util.ArrayList;
+
+@SuppressWarnings({"all", "warnings", "unchecked"})
+public class EsperLexer extends Lexer {
+    public static final int EOF=-1;
+    public static final int ASSIGN=4;
+    public static final int CHARACTER=5;
+    public static final int DECLARE=6;
+    public static final int DECREASING=7;
+    public static final int DIGITS=8;
+    public static final int DIV=9;
+    public static final int ELSE=10;
+    public static final int ELSEIF=11;
+    public static final int ENDFOR=12;
+    public static final int ENDIF=13;
+    public static final int ENDWHILE=14;
+    public static final int EQUALTO=15;
+    public static final int FOR=16;
+    public static final int GREATERTHAN=17;
+    public static final int GREATERTHANEQUAL=18;
+    public static final int IDENTIFIER=19;
+    public static final int IF=20;
+    public static final int INCREASING=21;
+    public static final int INFINITY=22;
+    public static final int LESSTHAN=23;
+    public static final int LESSTHANEQUAL=24;
+    public static final int MINUS=25;
+    public static final int MULT=26;
+    public static final int NEGATIVEINFINITY=27;
+    public static final int NULLITY=28;
+    public static final int PLUS=29;
+    public static final int PRINT=30;
+    public static final int SEMICOLON=31;
+    public static final int STRING=32;
+    public static final int VARINT=33;
+    public static final int VARSTRING=34;
+    public static final int VARTRANSREAL=35;
+    public static final int WHILE=36;
+    public static final int WHITESPACE=37;
+
+    // delegates
+    // delegators
+    public Lexer[] getDelegates() {
+        return new Lexer[] {};
+    }
+
+    public EsperLexer() {} 
+    public EsperLexer(CharStream input) {
+        this(input, new RecognizerSharedState());
+    }
+    public EsperLexer(CharStream input, RecognizerSharedState state) {
+        super(input,state);
+    }
+    public String getGrammarFileName() { return "C:\\Users\\Alex\\Dropbox\\EclipseWorkspace\\esper-compiler\\src\\compiler\\Esper.g"; }
+
+    // $ANTLR start "WHITESPACE"
+    public final void mWHITESPACE() throws RecognitionException {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.